src/plugins/intel_cpu/src/shape_inference/custom/multinomial.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018-2023 Intel Corporation
```

Comment:
Is it possible to use Core shape inference here? Custom shape inference is used if Core one cannot be use or has worse performance.

Reply:
I tried, but failed to create a tensor accessor from the map object. I left the initial impl idea in the comment for suggestions

Reply:
After recent changes, uses the Core shape inference as requested!